screenwriter/screenwriter.go

```diff
@@ -0,0 +1,60 @@
+package screenwriter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type ScreenWriter struct {
+	writer   io.StringWriter
+	previous string
+}
+
+func New(writer io.StringWriter) *ScreenWriter {
+	return &ScreenWriter{
+		writer: writer,
+	}
+}
+
+func (w *ScreenWriter) ClearPrevious() error {
+	if w.previous == "" {
+		return nil
+	}
+
+	// Get number of lines in previous
+	lines := strings.Count(w.previous, "\n")
+
+	// Handle the final line
+	finalLineLength := 0
+	splitLines := strings.Split(w.previous, "\n")
+	finalLine := splitLines[len(splitLines)-1]
+	finalLineLength = len(finalLine)
+	w.writer.WriteString(fmt.Sprintf("\u001b[%dD", finalLineLength))
```

Comment:
Sorry, this might be a bad question b/c I don't know Go so do ignore this if it's not helpful :)

I was wondering if there's a better way to indicate here that it's the exact byte `1B` that's at the beginning of this sequence, not the Unicode codepoint `U+001B`? Does Go have the `\x1b` syntax for representing bytes?

Reply:
It's both, isn't it? `U+001B` is a Unicode code point called "escape". And in UTF-8 it's exactly the same as `\x1b` (yes Go has both sequences).

Reply:
If the string is UTF-8 then this is fine. What the terminal wants here is the specific byte `1B`. In UTF-8, the codepoint U+001B encodes as the single byte `1B` which is what we want, but it may result in different bytes in other encodings.